controllers/resource/resource_generator.go

```diff
@@ -164,6 +179,59 @@ func (r *resourceGenerator) GetPodTemplate(doguResource *k8sv1.Dogu, dogu *core.
 	return podTemplate, nil
 }
 
+func getChownInitContainer(dogu *core.Dogu, doguResource *k8sv1.Dogu) (*corev1.Container, error) {
+	// Skip chown volumes with dogu-operator client because these are volumes from configmaps and read only.
+	filteredVolumes := filterVolumesWithClient(dogu.Volumes, doguOperatorClient)
+	if len(filteredVolumes) == 0 {
+		return nil, nil
+	}
+
+	initCommand := ""
+	for index, volume := range filteredVolumes {
+		uid, err := strconv.ParseInt(volume.Owner, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse owner id %s from volume %s: %w", volume.Owner, volume.Name, err)
+		}
+		gid, err := strconv.ParseInt(volume.Group, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse group id %s from volume %s: %w", volume.Group, volume.Name, err)
+		}
+
+		if uid > 0 && gid > 0 {
+			mkdirCommand := fmt.Sprintf("mkdir -p %s", volume.Path)
+			chownCommand := fmt.Sprintf("chown -R %s:%s %s", volume.Owner, volume.Group, volume.Path)
+			if index == 0 {
+				initCommand = fmt.Sprintf("%s && %s", mkdirCommand, chownCommand)
```

Comment:
Solely working on strings seems a bit off because you overwrite a variable again and again.
Did you consider working with string slices instead? I feel it would ber easier to append a new command in such a fasion and join the slice to a single with implements like `strings.Join()`